lucene/core/src/java/org/apache/lucene/search/KnnFloatVectorQuery.java

```diff
@@ -112,7 +112,14 @@ VectorScorer createVectorScorer(LeafReaderContext context, FieldInfo fi) throws
 
   @Override
   public String toString(String field) {
-    return getClass().getSimpleName() + ":" + this.field + "[" + target[0] + ",...][" + k + "]";
+    StringBuilder buffer = new StringBuilder();
+    buffer.append(getClass().getSimpleName() + ":");
+    buffer.append(this.field + "[" + target[0] + ",...]");
+    buffer.append("[" + k + "]");
+    if (this.filter != null) {
+      buffer.append(",filter:[" + this.filter + "]");
+    }
```

Comment:
```suggestion
    if (this.filter != null) {
      buffer.append("[" + this.filter + "]");
    }
```

I think that would be better.